File: src/main/java/io/codeka/gaia/repository/JobRepository.java
package io.codeka.gaia.repository;

import io.codeka.gaia.bo.Job;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * Repository for jobs
 */
@Repository
public interface JobRepository extends MongoRepository<Job, String>{

    List<Job> findAllByStackId(String stackId);

}

File: src/main/java/io/codeka/gaia/stacks/repository/JobRepository.kt
package io.codeka.gaia.stacks.repository

import io.codeka.gaia.stacks.bo.Job
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

/**
 * Repository for jobs
 */
@Repository
interface JobRepository : MongoRepository<Job, String> {

    fun findAllByStackIdOrderByStartDateTimeDesc(stackId: String): List<Job>

}
